back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/listener/ULPObservabilityListener.java

```diff
@@ -415,7 +415,18 @@ public void testEnded(String host) {
 			instanceCount--;
 			if (instanceCount == 0) {
 				LOG.info("No more test running, shutting down");
-
+				
+				if (!getSampleQueue().isEmpty()) {
+					LOG.info("The sample queue still not empty. {} samples remain to be consumed", getSampleQueue().size());
+					while(!getSampleQueue().isEmpty()) {
```

Comment:
This would trigger a 100% CPU:
- You could sleep within the loop for few ms and a max time as a solution